samples/bluetooth/iso_time_sync/src/controller_time_nrf52.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** This file implements controller time management for 52 Series devices
+ *
+ * As the controller clock is not directly accessible, the controller
+ * time is obtained using a mirrored RTC peripheral.
+ * To achieve microsecond accurate toggling, a timer peripheral is also used.
+ */
+
+#include <zephyr/kernel.h>
+#include <mpsl_clock.h>
+#include <nrfx_rtc.h>
+#include <nrfx_timer.h>
+#include <helpers/nrfx_gppi.h>
+#include <hal/nrf_egu.h>
+#include "iso_time_sync.h"
+
+static const nrfx_rtc_t app_rtc_instance = NRFX_RTC_INSTANCE(2);
+static const nrfx_timer_t app_timer_instance = NRFX_TIMER_INSTANCE(1);
+
+static volatile uint32_t num_rtc_overflows;
+
+static uint32_t offset_ticks_and_controller_to_app_rtc;
+
+static void rtc_isr_handler(nrfx_rtc_int_type_t int_type)
+{
+	if (int_type == NRFX_RTC_INT_OVERFLOW) {
+		num_rtc_overflows++;
+	}
+}
+
+static void unused_timer_isr_handler(nrf_timer_event_t event_type, void *ctx)
+{
+	ARG_UNUSED(event_type);
+	ARG_UNUSED(ctx);
+}
+
+static int32_t rtc_diff_get(void)
+{
+	uint32_t controller_ticks = nrf_rtc_counter_get(NRF_RTC0);
+	uint32_t app_ticks = nrf_rtc_counter_get(app_rtc_instance.p_reg);
+
+	return controller_ticks - app_ticks;
+}
+
+static int rtc_config(void)
+{
+	int ret;
+
+	const nrfx_rtc_config_t rtc_cfg = NRFX_RTC_DEFAULT_CONFIG;
+
+	ret = nrfx_rtc_init(&app_rtc_instance, &rtc_cfg, rtc_isr_handler);
+	if (ret - NRFX_ERROR_BASE_NUM) {
+		printk("Failed initializing RTC (ret: %d)\n", ret - NRFX_ERROR_BASE_NUM);
+		return ret - NRFX_ERROR_BASE_NUM;
+	}
+	IRQ_CONNECT(RTC2_IRQn, IRQ_PRIO_LOWEST, nrfx_rtc_2_irq_handler, NULL, 0);
```

Comment:
Could you use NRFX macro for obtaining interrupt line and handler?

Reply:
Can you provide an example of how this done? Most samples are not using that